### ISSUE TITLE
Can a struct have too many fields?

### ISSUE THREAD

**Issue by Pr0methean** [rust-lang/rust-clippy#16103]
### What it does

Currently `clippy::too_many_arguments` raises a warning, but there's no lint that warns about a struct or an enum variant having too many fields. That means that if a struct has 3 or 4 fields that a particular function needs, and they're otherwise unrelated, I have an incentive to pass in the entire struct instead. This encourages stamp coupling over data coupling, and makes borrow conflicts more likely, and makes missing (such as `BTreeMap::get_mut_many()`) and caveatful (such as `HashMap::get_mut_many()`'s runtime check for distinct keys) methods to work around these conflicts more painful.

Is there some opinionated reason to favor passing in the entire struct in such cases, or was it just an oversight?

### Advantage

Encourages devs not to borrow an entire struct in a function that only needs a few fields of that struct, and makes it possible to call that function even while other fields in the struct are mutably borrowed.

### Drawbacks

May hurt readability.

### Example

```rust
<code>
```

Could be written as:

```rust
<code>
```
